pkg/reconciler/identity/reconciler_test.go

```diff
@@ -74,7 +76,13 @@ func TestCreates(t *testing.T) {
 		expectedServiceAccount *corev1.ServiceAccount
 		wantCreates            []runtime.Object
 		wantErrCode            codes.Code
+		wantErr                error
+		addClusterName         bool
 	}{{
+		name:           "k8s service account doesn't exist, failed to get cluster name",
+		addClusterName: false,
+		wantErr:        fmt.Errorf(`failed to get cluster name: unable to get cluster name, please provide it by adding annotation "%s=$CLUSTER_NAME" to source`, duckv1alpha1.ClusterNameAnnotation),
```

Comment:
If this was not user facing, then I would probably do it this way. But because this is user facing, I can see the desire for it being painful to change it. I'm ok either way, but think I prefer hard coding to make it harder to accidentally change a user visible string.

Reply:
@grantr I removed this test, since it will show different error messages when running locally/running in prow. It is the same issue as the defaults test. If we didn't add clusterName annotation here, setDefaults() will set that annotation, and it can't take the testmedataClient in it. Thus, in prow, it can get the cluster name here...

Reply:
Ok thanks @Harwayne. Seems like this is affected by the same limitation mentioned above. @grac3gao please mention it in the issue from https://github.com/google/knative-gcp/pull/950#discussion_r422408427.

Reply:
Just added that. Issue #1037